### ISSUE TITLE
Remove the large interface definitions

### ISSUE THREAD

**Issue by steeling** [openservicemesh/osm#4931]
"Interfaces should be defined by where they are consumed, not by where they are instantiated" 

["The bigger the interface, the weaker the abstraction."](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=5m17s)

A k8s.Controller, doesn't need to know how it is used. Instead, whatever is consuming can define it's most limited interface, namely the methods *that client* uses.

For instance, once #4929 is complete, whatever may have used a configurator before, may define a small interface:

```go
type meshConfigGetter interface {
   GetMeshConfig() (*v1alpha1.MeshConfig, error)
}
```

[A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

This allows that method to be Dependency injected with other implementations, either for tests, or for other non-k8s control planes.

see https://go-proverbs.github.io/

**Comment by steeling** [comment]
I'm going to close this issue, and split it into multiple issues to allow parallelizing the work
